pkg/virt-launcher/virtwrap/nichotplug_test.go

```diff
@@ -324,7 +387,13 @@ func mockLibvirtClient(mockController *gomock.Controller, clientResult libvirtCl
 		mockClient.EXPECT().AttachDeviceFlags(gomock.Any(), gomock.Any()).Return(clientResult.expectedError)
 		return mockClient
 	}
-	mockClient.EXPECT().AttachDeviceFlags(gomock.Any(), gomock.Any()).Times(clientResult.expectedAttachedDevices).Return(nil)
+
+	if clientResult.expectedArgument == libvirtInterfaceLinkStateDown {
```

Comment:
Conditions in tests are a smell. This one may be resolved by passing in advance the expectations exactly and not an abstraction of it. You could use an abstraction in a form of a function that builds the expected data.

Bottom line, try to remove the condition please.